tools/cldr-apps/js/src/esm/cldrLoad.js

```diff
@@ -871,14 +900,18 @@ function trimNull(x) {
 function myLoad(url, message, handler, postData, headers) {
   const otime = new Date().getTime();
   console.log("MyLoad: " + url + " for " + message);
-  const errorHandler = function (err) {
-    console.log("Error: " + err);
-    notification.error({
-      message: `Could not fetch ${message}`,
-      description: `Error: ${err.toString()}`,
-      duration: 8,
-    });
-    handler(null);
+  const errorHandler = function (err, request) {
+    if (request.status === 404 && request.response?.code) {
+      handler(request.response); // pass through
```

Comment:
@btangmu this is why the handler is called with `json.code` set in case of a 404

Reply:
so in this case "json" is not really the json returned by the server, but json = request.response -- ? that's confusing for me

maybe room for improvement in the protocol, or maybe I'm still not getting it?

Reply:
it is the json returned by the server.  request.response is from the server.

Reply:
@btangmu  if you fetched `https://st.unicode.org/cldr-apps/api/voting/wxyz/page/BADPAGE` (but with the right surveytool-session header) you would get this 404 response:

```
{
  "code": "E_BAD_SECTION",
  "err": true,
  "message": "Error: E_BAD_SECTION"
}
```

Reply:
I think I get it now -- this code is `STError.code` filled in like this:

`return Response.status(404).entity(new STError(ErrorCode.E_BAD_SECTION)).build();`

That may seem obvious. Still, for easily-confused programmers like me, a comment relating `json.code` to `STError.code` would help, and even better would be to encapsulate that in cldrAjax -- something like `cldrAjax.getErrorCode(json)` which function could have a comment explaining the connection to STError on the back end.

It's not obvious in `if (!json || json.code)` that json.code is an error code -- after all, in general, a successful response could have `json.code`  meaning something completely different, right? For a while I was confusing it with `VoteAPI.RowResponse.Row.code`...

Reply:
the surrounding code now refers to STError.

Reply:
it could check `if (!json || json.err)` because err is always true in an STerror. maybe that would be clearer

Reply:
`myLoad` is called with many different handlers playing the role of `handler` -- will all of them grok the "pass through"?

Reply:
I think this line is only for this load

Reply:
OK, i didn't realize, `myLoad` is exported

> encapsulate that in cldrAjax

I think we should have less encapsulation here rather than more. It's already too many layers deep.